applications/FluidDynamicsHydraulicsApplication/custom_utilities/hydraulic_fluid_auxiliary_utilities.cpp

```diff
@@ -0,0 +1,322 @@
+{
+    std::size_t n_neg(0);
+    const std::size_t pts_number = rConditionDistancesVector.size();
+    for (std::size_t i_node = 0; i_node < pts_number; ++i_node)
+    {
+        if (rConditionDistancesVector[i_node] < 0.0)
+            n_neg++;
+    }
+    return n_neg == pts_number;
+}
+
+double HydraulicFluidAuxiliaryUtilities::InitialWaterDepth(ModelPart &rModelPart)
+{
+
+   //Determine the initial estimate for water depth by considering the average of the maximum and minimum coordinates.
+    KRATOS_WARNING("HydraulicFluidAuxiliaryUtilities") << "The water depth is assumed to be in the Z direction" << std::endl;
+    double max_value = -1e9;
+    double min_value = 1e9;
+    for (auto &r_node : rModelPart.Nodes())
+    {
+        if (r_node.Z() > max_value)
+        {
+            max_value = r_node.Z();
+        }
+        if (r_node.Z() < min_value)
+        {
+            min_value = r_node.Z();
+        }
+    }
+    double initial_water_depth = 0.5*std::abs(max_value-min_value);
+    return initial_water_depth;
+}
+
+void HydraulicFluidAuxiliaryUtilities::SetInletVelocity(
+    ModelPart &rModelPart,
+    double InletVelocity,
+    const Variable<double> &rDistanceVariable)
+{
+    block_for_each(rModelPart.Nodes(), [&](NodeType &rNode)
+    {
+        // TODO: The normal of the inlet conditions is currently assigned to the DISPLACEMENT variable, but it should be associated with another variable to prevent data superimposition
+        array_1d<double,3> inlet_norm = rNode.GetValue(DISPLACEMENT);
+
+        const double n_norm = norm_2(inlet_norm);
+        // Orient the velocity vector in the opposite direction of the outer normal to ensure it is directed inwards.
+
+        if (n_norm > 1.0e-12)
+        {
+            inlet_norm /= -n_norm;
+        }
+        else
+        {
+            KRATOS_WARNING("SetInletVelocity") << "Node " << rNode.Id() << " INLET_NORMAL is close to zero." << std::endl;
+            inlet_norm /= -1.0;
+        }
+        //  Inlet velocity vector.
+        array_1d<double,3> inlet_velocity = inlet_norm * InletVelocity;
+
+        //  Assign the velocity vector to each node representing water in the inlet condition and fix its value
+        if (rNode.GetValue(rDistanceVariable) < 0.0)
+        {
+            rNode.FastGetSolutionStepValue(VELOCITY) =  inlet_velocity;
+            rNode.Fix(VELOCITY_X);
+            rNode.Fix(VELOCITY_Y);
+            rNode.Fix(VELOCITY_Z);
+        }
+        else{
+            // The air velocity in the inlet node is assumed to be null.
+            rNode.FastGetSolutionStepValue(VELOCITY_X) = 0.0;
+            rNode.FastGetSolutionStepValue(VELOCITY_Y) = 0.0;
+            rNode.FastGetSolutionStepValue(VELOCITY_Z) = 0.0;
+            rNode.Fix(VELOCITY_X);
+            rNode.Fix(VELOCITY_Y);
+            rNode.Fix(VELOCITY_Z);
+        }
+    });
+}
+void HydraulicFluidAuxiliaryUtilities::FreeInletVelocity(ModelPart& rModelPart)
```

Comment:
```suggestion
void HydraulicFluidAuxiliaryUtilities::FreeInlet(ModelPart& rModelPart)
```
Name is misleading as you are freeing both velocity and distance. I'd keep `FreeInlet` just in case we need to modify it in the future.

Reply:
The name is modified in both files.